### PR TITLE
CAMEL-14832: Data format for SWIFT financial message conversion support

### PR DESCRIPTION
**Pull request by essobedo**
Fix for https://issues.apache.org/jira/browse/CAMEL-14832

## Motivation

Able to support SWIFT financial message conversion with Prowide Core library. This library has Apache License 2.0. Available from https://github.com/prowide/prowide-core.

## Modifications

* Add a new module called `camel-swift`
* Add 2 data formats which are `swiftMt` and `swiftMx` for respectively the SWIFT MT messages and the SWIFT MX messages.

### CODE REVIEW

File: core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMtDataFormat.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import org.apache.camel.model.DataFormatDefinition;
+import org.apache.camel.spi.Metadata;
+
+/**
+ * Encode and decode SWIFT MT messages.
+ */
+@Metadata(firstVersion = "3.20.0", label = "dataformat,transformation,swift,mt", title = "SWIFT MT")
```
Comment (davsclaus) [inline core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMtDataFormat.java:30]: Label = mt is maybe too far. eg users will find this via swift

"dataformat,transformation,swift"

File: core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMxDataFormat.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+
+import org.apache.camel.model.DataFormatDefinition;
+import org.apache.camel.spi.Metadata;
+
+/**
+ * Encode and decode SWIFT MX messages.
+ */
+@Metadata(firstVersion = "3.20.0", label = "dataformat,transformation,swift,mx", title = "SWIFT MX")
```
Comment (davsclaus) [inline core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMxDataFormat.java:31]: Same as prev comment

**Review by davsclaus — APPROVED** [review]

File: core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMtDataFormat.java (hunk 1 above)
Comment (essobedo) [reply core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMtDataFormat.java:30]: Just to make it clear, you would rather prefer that I remove the tags mt and mx, is that correct?

Comment (davsclaus) [reply core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/SwiftMtDataFormat.java:30]: Yes correct - as we should have a limited set of labels. If you search for MT then you find that anyway via the name. A label is like a common label for N+ - not a unique label.

**Comment by essobedo** [comment]
The checkstyle build fails because no snapshot of `camel-swift` has been published  yet

**Pull request merged**